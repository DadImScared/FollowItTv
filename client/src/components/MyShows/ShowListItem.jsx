
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Button from '@material-ui/core/Button';
import ListItem from '@material-ui/core/ListItem';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';

class ShowListItem extends React.PureComponent {

  unFollow = () => {
    const { show } = this.props;
    this.props.unFollow(show);
  };

  render() {
    const { show } = this.props;
    return (
      <ListItem component={Link} to={`/show/${show.id}`} button divider={true}>
        <ListItemText style={{ flex: '0.7 1 auto' }} primary={show.name} />
        <ListItemSecondaryAction>
          <Button onClick={this.unFollow}>
            delete
          </Button>
        </ListItemSecondaryAction>
      </ListItem>
    );
  }
}

ShowListItem.propTypes = {
  show: PropTypes.object.isRequired,
  unFollow: PropTypes.func.isRequired
};

export default ShowListItem;
