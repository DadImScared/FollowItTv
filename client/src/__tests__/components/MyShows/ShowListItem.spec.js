
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';

import Button from '@material-ui/core/Button';

import ShowListItem from '../../../components/MyShows/ShowListItem';

describe('ShowListItem', () => {
  let props;

  beforeEach(() => {
    props = {
      show: {
        id: 1,
        name: 'breaking bad'
      },
      unFollow: sinon.spy()
    };
  });

  it('should render', () => {
    const wrapper = shallow(<ShowListItem {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call unFollow on button click', () => {
    const wrapper = shallow(<ShowListItem {...props} />);
    wrapper.find(Button).simulate('click');
    expect(props.unFollow.calledOnce).toEqual(true);
  });
});
