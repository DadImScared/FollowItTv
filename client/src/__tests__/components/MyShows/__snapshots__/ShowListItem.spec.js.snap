// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowListItem should render 1`] = `
<WithStyles(ListItem)
  button={true}
  component={[Function]}
  divider={true}
  to="/show/1"
>
  <WithStyles(ListItemText)
    primary="breaking bad"
    style={
      Object {
        "flex": "0.7 1 auto",
      }
    }
  />
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(Button)
      onClick={[Function]}
    >
      delete
    </WithStyles(Button)>
  </WithStyles(ListItemSecondaryAction)>
</WithStyles(ListItem)>
`;
